base/linalg/lapack.jl

```diff
@@ -1205,8 +1205,8 @@ for (gelsd, gelsy, elty) in
             end
             newB = [B; zeros($elty, max(0, n - size(B, 1)), size(B, 2))]
             s     = similar(A, $elty, min(m, n))
-            rcond = convert($elty, rcond)
-            rnk   = Vector{BlasInt}(1)
+            rcond = Ref{$elty}(rcond)
```

Comment:
AFAIU this one would always be inferred as `::Union{Ref{$elty}, <orig. rcond type>}`, whereas `convert($elty, rcond)` gave `::Union{$elty, <orig. rcond type>}` only when `$elty !== <orig. rcond type>`.
Maybe introduce a new var ("**r**rcond") for a better type inference?